### ISSUE TITLE
[bug] Programatically add menu items to left menu

### ISSUE THREAD

**Issue by Mihai-B** [mika-el/angular-admin-lte#20]
First, thank you for this implementation, it is really nicely done.

I am trying to add a menu with submenu on the left sidebar like so:
```
menuItems.push({
      label: 'abc', iconClasses: 'fa fa-newspaper-o',
      children: [ { label: 'X', route: 'contact' },  { label: 'Y', route: 'contact' }  ]
    });
  this.layoutStore.setSidebarLeftMenu(menuItems);
```
If I add this menu item to the layout store, then if you click on the apc to expand the menu, the page just reloads.

I think this may be a bug

**Comment by Mihai-B** [comment]
Ok I found out how to.You can set it on the LayoutStore.

GitHub should have a delete function if there are no replies to a issue so I can delete my stupid questions sometimes.

Thnx again for the admin lte angular implementation. Really really professionally made.

**Comment by mika-el** [comment]
Hi,

Thanks for the feedback :)

You need to change Array ref: 

```
menuItems.push({
  label: 'abc', iconClasses: 'fa fa-newspaper-o', children:  [ 
    {label: 'X', route: 'contact'},  {label: 'Y', route: 'contact'}  
  ]
});

menuItems = [... menuItems];

this.layoutStore.setSidebarLeftMenu(menuItems);
```

**Comment by Mihai-B** [comment]
Hi mika-el, 
I don't think I follow.

I create my array like this: let menuItems = [];

But still, I tried what you suggested and it is the same.

**Comment by mika-el** [comment]
I created a codesanbox https://codesandbox.io/s/nww4xj4l7l and It works here.
Can you check it and if you still can't make it work, create a sandbox, I'll help you to find out why.

**Comment by Mihai-B** [comment]
@mika-el  the code in codesanbox does not work...
On the codesanbox you provided, expend the Layout menu or the Boxs menu. you will see that it is not expanding

**Comment by mika-el** [comment]
I thought you were talking about menu not updating changes, sorry :)
bug fixed, update angular-admin-lte to 1.2.4.

**Comment by Mihai-B** [comment]
I confirm, issue is fixed. Thnx for the help.
